pkg/operator/ceph/object/bucket/s3-handlers.go

```diff
@@ -0,0 +1,371 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bucket
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/yard-turkey/lib-bucket-provisioner/pkg/provisioner/api/errors"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+// S3Agent wraps the s3.S3 structure to allow for wrapper methods
+type S3Agent struct {
```

Comment:
is this s3 agent used to do any work in AWS? or is it only used as a client to speak the s3 protocol to RGW buckets?

Reply:
That's correct. Specifically, it's necessary to add, edit, and remove bucket policies for brownfield access.